packages/authgear-core/src/base64.ts

```diff
@@ -1,43 +1,18 @@
-import { _ByteArray } from "./types";
+import { encode, decode } from "base64-arraybuffer";
 
-function uint6ToB64URL(nUint6: number) {
-  return nUint6 < 26
-    ? nUint6 + 65
-    : nUint6 < 52
-    ? nUint6 + 71
-    : nUint6 < 62
-    ? nUint6 - 4
-    : nUint6 === 62
-    ? 45
-    : nUint6 === 63
-    ? 95
-    : 65;
-}
-
-function base64URLEncArr(aBytes: _ByteArray) {
-  const eqLen = (3 - (aBytes.length % 3)) % 3;
-  let sB64Enc = "";
-  let nUint24 = 0;
-  for (let nIdx = 0; nIdx < aBytes.length; nIdx++) {
-    const nMod3 = nIdx % 3;
-    nUint24 |= aBytes[nIdx] << ((16 >>> nMod3) & 24);
-    if (nMod3 === 2 || aBytes.length - nIdx === 1) {
-      sB64Enc += String.fromCharCode(
-        uint6ToB64URL((nUint24 >>> 18) & 63),
-        uint6ToB64URL((nUint24 >>> 12) & 63),
-        uint6ToB64URL((nUint24 >>> 6) & 63),
-        uint6ToB64URL(nUint24 & 63)
-      );
-      nUint24 = 0;
-    }
-  }
-
-  return eqLen === 0 ? sB64Enc : sB64Enc.substring(0, sB64Enc.length - eqLen);
+/**
+ * @internal
+ */
+export function _base64URLEncode(arrayBuffer: ArrayBuffer): string {
+  return encode(arrayBuffer)
+    .replace(/\//g, "_")
+    .replace(/\+/g, "-")
+    .replace(/=+$/, "");
 }
 
 /**
  * @internal
  */
-export function _encodeBase64URLFromByteArray(bytes: _ByteArray): string {
-  return base64URLEncArr(bytes);
+export function _base64URLDecode(s: string): ArrayBuffer {
+  return decode(s.replace(/-/g, "+").replace(/_/g, "/"));
```

Comment:
Is it necessary to add back tailing `=` like what we did in iOS sdk?

Reply:
I have checked the source code of the library we are using. [The padding is actually ignored.](https://github.com/niklasvh/base64-arraybuffer/blob/master/lib/base64-arraybuffer.js#L44) so I did not add the padding.

Reply:
ok, that's great. Could we put comments here so we won't confuse in the future? :)